applications/eigen_solvers_application/CMakeLists.txt

```diff
@@ -0,0 +1,72 @@
+# 
+#   KRATOS _______
+#         / ____(_)___ ____  ____
+#        / __/ / / __ `/ _ \/ __ \
+#       / /___/ / /_/ /  __/ / / /
+#      /_____/_/\__, /\___/_/ /_/ SolversApplication
+#              /____/
+# 
+#   Author: Thomas Oberbichler
+# 
+
+set( CMAKE_INCLUDE_CURRENT_DIR ON )
+
+message( "**** configuring KratosEigenSolversApplication ****" )
+
+if( NOT DEFINED EIGEN_ROOT )
+	message( FATAL_ERROR "EIGEN_ROOT not defined" )
+endif()
+
+if( MSVC )
+	set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /Ox" )
+else()
+	set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O3 -m64 -L$ENV{MKLROOT}/lib/intel64 -Wl,--no-as-needed -lpthread -lm -ldl" )
+endif()
+
+add_definitions( -DNDEBUG -DEIGEN_DEFAULT_TO_ROW_MAJOR -DEIGEN_MPL2_ONLY )
+
+include_directories(
+	${CMAKE_SOURCE_DIR}/kratos
+	${EIGEN_ROOT}
+)
+
+if( DEFINED ENV{MKLROOT} )
+	message( "MKLROOT = $ENV{MKLROOT}" )
+	include_directories( $ENV{MKLROOT}/include )
+	add_definitions( -DEIGEN_USE_MKL_ALL )
+else()
+	message( "MKLROOT not defined" )
+endif()
+
+## generate variables with the sources
+set( KRATOS_EIGENSOLVERS_APPLICATION_SOURCES     
+	${CMAKE_CURRENT_SOURCE_DIR}/eigen_solvers_application.cpp 
+	${CMAKE_CURRENT_SOURCE_DIR}/custom_python/eigen_solvers_python_application.cpp
+	${CMAKE_CURRENT_SOURCE_DIR}/custom_python/add_custom_solvers_to_python.cpp
```

Comment:
I would rename these files to sth with `linear` in it to make it clear what kind of solver it is, e.g.  `add_custom_linear_solvers_to_python`

Reply:
Then also the directory should be renamed. I try to keep it simple.